src/services/rabbitmqRelayPlugin.ts

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+import amqplib from 'amqplib';
+import type { Channel, ChannelModel } from 'amqplib';
+import type { LoggerService } from '@tazama-lf/frms-coe-lib';
+import type { Apm } from '@tazama-lf/frms-coe-lib/lib/services/apm';
+import { additionalEnvironmentVariables, type Configuration } from '../config';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config/processor.config';
+import type { ITransportPlugin } from '@tazama-lf/frms-coe-lib/lib/interfaces/relay-service/ITransportPlugin';
+import fs from 'fs';
+
+export default class RabbitMQRelayPlugin implements ITransportPlugin {
+  private amqpConnection?: ChannelModel;
+  private amqpChannel?: Channel;
+  private loggerservice?: LoggerService;
+  private apm?: Apm;
+  private readonly configuration: Configuration;
+
+  constructor() {
+    this.configuration = validateProcessorConfig(additionalEnvironmentVariables) as Configuration;
+  }
+
+  /**
+   * Initializes the RabbitMQ connection for the relay plugin.
+   *
+   * This method establishes a connection to RabbitMQ using either TLS or plain connection
+   * based on the environment configuration. In non-development environments with TLS
+   * certificate available, it creates a secure connection. Otherwise, it creates a
+   * standard connection and channel.
+   * @param loggerService - Optional logger service for logging operations.
+   * @param apm - Optional APM service for performance monitoring.
+   * @returns A Promise that resolves when the connection is successfully established
+   * @throws {Error} Throws an error if the connection to RabbitMQ fails
+   *
+   * @remarks
+   * - In production environments with RABBITMQ_TLS_CA configured, uses TLS connection
+   * - In development or when TLS is not configured, uses standard connection
+   * - Creates an AMQP channel only for non-TLS connections
+   * - Logs connection status and server properties for debugging
+   */
+  async init(loggerService?: LoggerService, apm?: Apm): Promise<void> {
+    this.loggerservice = loggerService;
+    this.apm = apm;
+    this.loggerservice?.log('RabbitMQ Relay Plugin initialized', RabbitMQRelayPlugin.name);
+    try {
+      if (this.configuration.nodeEnv !== 'dev') {
+        if (!this.configuration.RABBITMQ_TLS_CA) {
+          throw new Error('TLS certificate (RABBITMQ_TLS_CA) is required in non-development environments');
+        }
+        const tlsOptions = {
+          ca: fs.readFileSync(this.configuration.RABBITMQ_TLS_CA, 'utf-8'),
+        };
+        this.amqpConnection = await amqplib.connect(this.configuration.DESTINATION_TRANSPORT_URL, tlsOptions);
+        this.amqpChannel = await this.amqpConnection.createChannel();
```

Comment:
We have two branches doing the same kind of work. I see "connect()` on the two branches as well. If `tlsOptions` can be undefined, can we not use that route instead?

```ts
let tlsOptions: TlsOptions|undefined;
if (!dev) {
  tlsOptions = /* */;
}
this.amqpConnection = await amqplib.connect("", tlsOptions);
```

Reply:
This way, we only call the connect once